Makefile

```diff
@@ -1,25 +1,66 @@
+ROOT_DIR := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+OUTPUT_DIR := $(ROOT_DIR)/output
+BUILD_DIR := $(OUTPUT_DIR)/build
+IMAGE_DIR := $(OUTPUT_DIR)/images
+INITRAMFS_DIR := $(OUTPUT_DIR)/initramfs
 
-all: buildroot/Makefile buildroot/.config
-	make -C buildroot
+BR_INIT_ENV := BR2_EXTERNAL=$(ROOT_DIR)
 
+
+.PHONY: all
+all: $(IMAGE_DIR)/rootfs.img $(IMAGE_DIR)/uImage $(IMAGE_DIR)/am335x-kno_txt.dtb
+
+.PHONY: clean
 clean:
-	BR2_EXTERNAL=.. make -C buildroot clean
+	rm -rf $(OUTPUT_DIR)
+
+.PHONY: prepare
+prepare: $(BUILD_DIR)/rootfs/.config $(BUILD_DIR)/initramfs/.config
+
+.PHONY: source
+source: rootfs-source initramfs-source
+
+.PHONY: rootfs-source
+rootfs-source: $(BUILD_DIR)/rootfs/.config
+	$(MAKE) -C $(BUILD_DIR)/rootfs source
 
-buildroot/Makefile:
+.PHONY: initramfs-source
+initramfs-source: $(BUILD_DIR)/initramfs/.config
+	$(MAKE) -C $(BUILD_DIR)/initramfs source
+
+$(BUILD_DIR)/rootfs/.config: $(ROOT_DIR)/buildroot/Makefile
+	mkdir -p $(BUILD_DIR)/rootfs
+	$(BR_INIT_ENV) $(MAKE) O=$(BUILD_DIR)/rootfs -C $(ROOT_DIR)/buildroot fischertechnik_TXT_rootfs_defconfig
+
+$(BUILD_DIR)/initramfs/.config: $(ROOT_DIR)/buildroot/Makefile
+	mkdir -p $(BUILD_DIR)/initramfs
+	$(BR_INIT_ENV) $(MAKE) O=$(BUILD_DIR)/initramfs -C $(ROOT_DIR)/buildroot fischertechnik_TXT_initramfs_defconfig
+
+$(ROOT_DIR)/buildroot/Makefile:
 	git submodule update --init buildroot
 
-CONFIG_DEPENDS = \
-  .gitmodules \
-  board/fischertechnik/TXT/tisdk_am335x-fischertechnik_txt_defconfig
+$(IMAGE_DIR)/rootfs.img $(IMAGE_DIR)/uImage $(IMAGE_DIR)/am335x-kno_txt.dtb:
+	$(MAKE) rootfs
+	mkdir -p $(IMAGE_DIR)
+	cp $(BUILD_DIR)/rootfs/images/rootfs.squashfs $(IMAGE_DIR)/rootfs.img
+	cp $(BUILD_DIR)/rootfs/images/uImage $(IMAGE_DIR)/uImage
+	cp $(BUILD_DIR)/rootfs/images/device_tree.dtb $(IMAGE_DIR)/am335x-kno_txt.dtb
+
+.PHONY: rootfs
+rootfs: $(BUILD_DIR)/rootfs/.config $(INITRAMFS_DIR)/initramfs.cpio
+	$(MAKE) -C $(BUILD_DIR)/rootfs
 
-buildroot/.config: $(CONFIG_DEPENDS) buildroot/Makefile
-	BR2_EXTERNAL=.. make -C buildroot fischertechnik_TXT_defconfig
+$(INITRAMFS_DIR)/initramfs.cpio:
+	$(MAKE) initramfs
```

Comment:
Can't this be modeled as dependency to initramfs? Probably not because it should not be executed in parallel to other dependencies.